### ISSUE TITLE
Next steps

### ISSUE THREAD

**Issue by MilesMcBain** [cranchange/cran_change.org#31]
We seem to have converged on a path for now that involves making direct approaches to parties with interests in this issue to try to minimise collateral damage.

* As @gmbecker has suggested I think we need to reach out to CRAN
* I have contacted 2 members of R Core, but received no reply. I'll try one more that has been suggested
* @wlandau has contacted the Linux Foundation
* We need to make an approach to some members of the R foundation
* Also the R Consortium
* Early on @llrs made representations to R Forwards. Is it time to follow up on that?

Each of these communications will need to be tailored to the org and the people for best chance of getting anywhere. We'll need to coordinate our communications a little so that we don't accidentally contact the same people separately.

I think the ask itself is pretty simple, along the lines of what @wlandau wrote to the Linux Foundation:https://github.com/cranchange/cran_change.org/issues/27#issuecomment-878694203

We are concerned about CRAN's policies and administration etc, we have a list of proposed reforms that will make CRAN more robust and save administrators and contributors time etc,  and are looking for:
* Feedback on our reforms
* Advice as to how best to realise them with minimal harm
* Organisational or personal support if they are willing to give it, including lobbying and or brokering engagement with CRAN
  - OR direction toward people or groups who may be well placed to do this.

The approach to CRAN itself is a bit different and probably the most challenging. Any objections to drafting that together here as we have done for the reforms?

**Comment by MilesMcBain** [comment]
I created a private thread for communication coordination visible to @cranchange/authors.

**Comment by wlandau** [comment]
Thanks, @MilesMcBain! Would it also help to create a private repo under https://github.com/cranchange with these communications?

**Comment by llrs** [comment]
I do not represent this organization on R Forwards/R Working contributing to R group (or any other group). I will only speak by myself and my concerns about the governance and future of the R language and ecosystem. I don't feel I have any right to represent the other authors but I will ask again for feedback and advice. As far as I know, receiving and giving support works on the individual level. I anyone want to join the slack channel or join the meeting I'm sure you'll be well received https://forwards.github.io/rcontribution/slack. 

The normal process to collaborate with the R core is to comment something and if it is of their interest continue working on it with (private) feedback from one or multiple R core members. Perhaps the current approach is the wrong way: First we could offer our help to the CRAN team to start with something they desire but they don't have time or resources to do but that someone else could do. Later on, when there is a better communication and understanding perhaps the **suggestions** will be better tailored and better received.

**Comment by MilesMcBain** [comment]
@wlandau could you see the thread I started? I don't mind a repo, but just checking.

**Comment by wlandau** [comment]
Yes, I can see the thread. And I am all for whatever is most expedient (repo vs thread).

**Comment by bschneidr** [comment]
As someone frustrated with recent CRAN interactions of the nature specifically outlined in this letter, I'm curious what has happened as a result of the letter and conversations. Is there any way to learn what came of this?

**Comment by MilesMcBain** [comment]
We were convinced to try working with the Linux Foundation/ R Consortium to try and get a dialogue going with CRAN. That resulted in formation of this working group: https://github.com/RConsortium/r-repositories-wg

Have a read through the minutes if you want to get a feel for the progress that’s been made - little if any in my assessment.

**Comment by llrs** [comment]
@bschneidr The repository working group was started and there we have been meeting to discuss multiple approaches to the problems with CRAN. We had some meetings with the CRAN team (and R core members) were we provided some questions and feedback to the CRAN team for their consideration (some of which we haven't heard back yet). Besides, the working group has been making some very slow progress, recently these include:

1. There is the possibility that CRAN will use containers to test the packages, solving security concerns from CRAN and reproducibility concerns from package developers. There are ongoing conversations about how to get to that.
2. The CRAN team is aware of the messaging problems, making them work more and not being clear enough to developers, recently we started working to clarify the checks messages provided.

I hope this helps and I too suggest you to read the minutes of the working group (but in my opinion they don't represent well the content of the meetings).

**Comment by DataStrategist** [comment]
Might I suggest the cran team provide an anonymized dataset of their thousands of conversations with devs? Could create an interesting corpus for question answering that devs could use to troubleshoot issues prior to submission. Eventually it could even be part of the cran check, in order to provide more helpful and meaningful tips about errors. This would make the submission process easier for new devs and reduce volume for the cran team.

**Comment by bschneidr** [comment]
Thanks @MilesMcBain and @llrs for explaining and sharing those updates. I think it's a great idea for CRAN to use containers, particularly so that those can be published in a way that helps users reproduce or understand CRAN's testing configurations. If CRAN used containers and published those transparently, it would go a long way towards helping people reproduce the more obscure and pesky errors that keep package devs awake at night for two weeks. 

Improving messaging in checks is definitely helpful too.

**Comment by llrs** [comment]
@DataStrategist This is the purpose of the [r-package-devel mailing list](https://stat.ethz.ch/mailman/listinfo/r-package-devel). There is already a big corpus of questions and answers, from the community and the CRAN team. I often search in the archives for problems I get via: `site:https://stat.ethz.ch/pipermail/r-package-devel/ my problem`.
